sky/backends/cloud_vm_ray_backend.py

```diff
@@ -2806,7 +2821,8 @@ def _teardown(self,
             with filelock.FileLock(
                     lock_path,
                     backend_utils.CLUSTER_STATUS_LOCK_TIMEOUT_SECONDS):
-                success = self.teardown_no_lock(handle, terminate, purge)
+                success = self.teardown_no_lock(
+                    handle, terminate, purge, refresh_cluster_status=not purge)
```

Comment:
some comments on why set `refresh_cluster_status` to `not purge`

Reply:
Done.